pkg/utils/utils_test.go

```diff
@@ -2,26 +2,109 @@ package utils
 
 import (
 	"bytes"
+	"database/sql"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 // TestCompressDecompressGZIP tests both CompressGZIP and DecompressGZIP functions.
 func TestCompressDecompressGZIP(t *testing.T) {
 	mockData := []byte(`{"test": "This is a JSON file"}`)
 
-	// Compress the data
+	// Compress data
 	compressedData, err := CompressGZIP(mockData)
 	if err != nil {
 		t.Errorf("Error compressing data: %v", err)
 	}
 
-	// Decompress the data to verify
+	// Decompress data to verify
 	decompressedData, err := DecompressGZIP(compressedData)
 	if err != nil {
 		t.Errorf("Error decompressing data: %v", err)
 	}
 
 	if !bytes.Equal(mockData, decompressedData) {
-		t.Errorf("Decompressed data does not match original data:\n Expected: %v, Got: %v", mockData, decompressedData)
+		t.Fatalf("Decompressed data does not match original data:\n Expected: %v\n Got: %v", string(mockData), string(decompressedData))
+	}
+}
+
+// TestCompressWhenNeeded tests CompressWhenNeeded function
+func TestCompressWhenNeeded(t *testing.T) {
+	tests := []struct {
+		name           string
+		data           []byte
+		expectCompress bool
+	}{
+		{
+			name:           "Should not compress",
+			data:           []byte(`{"test": false}`),
+			expectCompress: false,
+		},
+		{
+			name: "Should compress",
+			data: []byte(`{
+				"name": "This is a test",
+				"test": "This should be compressed",
+				"additional_data": "Aliqua enim officia eiusmod ad. Officia cillum dolore occaecat consectetur amet dolore commodo adipisicing ut ut. Sit eiusmod aliquip occaecat laborum aliquip qui duis ut elit duis. Eiusmod ullamco elit Lorem nostrud consequat adipisicing quis cupidatat. Aliqua nulla ad aliqua exercitation amet ea excepteur nisi anim officia in voluptate commodo exercitation. Minim cupidatat proident aliquip minim officia id occaecat ea est Lorem nulla irure nulla excepteur."
+			}`),
+			expectCompress: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			compressedData, compression, err := CompressWhenNeeded(tt.data)
+			if err != nil {
+				t.Fatalf("Error determining whether or not to compress data: %v", err)
+			}
+
+			if tt.expectCompress {
+				assert.NotNil(t, compression)
+				assert.Equal(t, "gzip", *compression)
```

Comment:
Possible should also assert here that the returned compressedData != tt.data, e.g.

```suggestion
				assert.Equal(t, "gzip", *compression)
				assert.NotEqual(t, tt.data, compressedData)
```